internal/benchmarks/trace/clientServer_test.go

```diff
@@ -0,0 +1,446 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"testing"
+
+	"google.golang.org/grpc"
+)
+
+// Start: 0. PlainClient-PlainServer
+func BenchmarkPlainClientPlainServer1QPS(b *testing.B) {
+	benchmarkPlainClientPlainServer(b, 1)
+}
+
+func BenchmarkPlainClientPlainServer10QPS(b *testing.B) {
+	benchmarkPlainClientPlainServer(b, 10)
+}
+
+func BenchmarkPlainClientPlainServer100QPS(b *testing.B) {
+	benchmarkPlainClientPlainServer(b, 100)
+}
+
+func BenchmarkPlainClientPlainServer1000QPS(b *testing.B) {
+	benchmarkPlainClientPlainServer(b, 1000)
+}
+
+func benchmarkPlainClientPlainServer(b *testing.B, qps int) {
+	conn, err := grpc.Dial(plainServerAddr, grpc.WithInsecure())
+	if err != nil {
+		b.Fatalf("Creating traced gRPC connection: %v", err)
+	}
+	defer conn.Close()
+	runWithConn(b, conn, qps)
+}
+// End: 0. PlainClient-PlainServer
+
+// Start: 1. NoTraceNoStatsClient-NoTraceNoStatsServer
+func BenchmarkNoTraceNoStatsClientNoTraceNoStatsServer1QPS(b *testing.B) {
+	benchmarkNoTraceNoStatsClientNoTraceNoStatsServer(b, 1)
+}
+
+func BenchmarkNoTraceNoStatsClientNoTraceNoStatsServer10QPS(b *testing.B) {
+	benchmarkNoTraceNoStatsClientNoTraceNoStatsServer(b, 10)
+}
+
+func BenchmarkNoTraceNoStatsClientNoTraceNoStatsServer100QPS(b *testing.B) {
+	benchmarkNoTraceNoStatsClientNoTraceNoStatsServer(b, 100)
+}
+
+func BenchmarkNoTraceNoStatsClientNoTraceNoStatsServer1000QPS(b *testing.B) {
+	benchmarkNoTraceNoStatsClientNoTraceNoStatsServer(b, 1000)
+}
+
+func benchmarkNoTraceNoStatsClientNoTraceNoStatsServer(b *testing.B, qps int) {
+	conn, err := grpc.Dial(noTraceNoStatsAddr, noTraceNoStatsClient, grpc.WithInsecure())
+	if err != nil {
+		b.Fatalf("Creating traced gRPC connection: %v", err)
+	}
+	defer conn.Close()
+	runWithConn(b, conn, qps)
+}
+
+// End: 1. NoTraceNoStatsClient-NoTraceNoStatsServer
+
+// Start: 2. NoTraceNoStatsClient-NoTraceYesStatsServer
+func BenchmarkNoTraceNoStatsClientNoTraceYesStatsServer1QPS(b *testing.B) {
+	benchmarkNoTraceNoStatsClientNoTraceYesStatsServer(b, 1)
+}
+
+func BenchmarkNoTraceNoStatsClientNoTraceYesStatsServer10QPS(b *testing.B) {
+	benchmarkNoTraceNoStatsClientNoTraceYesStatsServer(b, 10)
+}
+
+func BenchmarkNoTraceNoStatsClientNoTraceYesStatsServer100QPS(b *testing.B) {
```

Comment:
I agree, the reason they are like that was because they are quite many i.e. 64 and so needed a common nomenclature to be able to disambiguate between them.